configs/cache_partitioning/cache_partitioning.py

```diff
@@ -0,0 +1,214 @@
+    max_ticks=2000000000,
+)
+
+
+def capacityAllocation(capacity_str):
+    """
+    Verify that Max Capacity partitioning policy has been provided with a suitable
+    configuration
+    """
+    capacity = float(capacity_str)
+    assert (
+        capacity >= 0
+    ), "Max Capacity Policy needs allocation in range [0, 1]"
+    assert (
+        capacity <= 1
+    ), "Max Capacity Policy needs allocation in range [0, 1]"
+
+    return capacity
+
+
+def wayAllocation(way_str):
+    """
+    Verify that Way partitioning policy has been provided with a suitable
+    configuration
+    """
+    way_alloc = int(way_str)
+    assert way_alloc >= 0, "Way Policy needs positive number of ways"
+
+    return way_alloc
+
+
+def generatePartPolicy(args):
+    """
+    Generate Partitioning Policy object based on provided arguments
+    """
+    assert args.policy in [
+        "way",
+        "max_capacity",
+    ], "Only support generating way and max_capacity policies"
+
+    if args.policy == "way":
+        allocated_ways = [way for way in range(0, args.way_allocation)]
+        allocation = WayPolicyAllocation(partition_id=0, ways=allocated_ways)
+
+        return WayPartitioningPolicy(allocations=[allocation])
+
+    if args.policy == "max_capacity":
+        return MaxCapacityPartitioningPolicy(
+            partition_ids=[0], capacities=[args.capacity_allocation]
+        )
+
+
+def configSystem():
+    """
+    Configure base system and memory
+    """
+
+    system = System(membus=IOXBar(width=128))
+    system.clk_domain = SrcClockDomain(
+        clock=STATIC_OPTIONS.system_frequency,
+        voltage_domain=VoltageDomain(),
+    )
+
+    # add memory
+    mem_range = AddrRange("128MB")
+    system.mem_ranges = [mem_range]
+
+    # Memory configuration
+    config_mem(STATIC_OPTIONS, system)
+    system.mem_ctrls[0].latency = STATIC_OPTIONS.request_latency
+    system.mem_ctrls[0].bandwidth = STATIC_OPTIONS.mem_bandwidth
+
+    return system
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+parser.add_argument(
+    "--policy",
+    default="way",
+    choices=["way", "max_capacity"],
+    help="This option defines which Cache Partitioning Policy to use for "
+    "the system cache",
+)
+
+parser.add_argument(
+    "--capacity-allocation",
+    type=capacityAllocation,
+    default=0.5,
+    help="The amount of the cache to partition to the default PartitionID "
+    "when using Max Capacity Cache Partitioning Policy in [0,1] range",
+)
+
+parser.add_argument(
+    "--way-allocation",
+    type=wayAllocation,
+    default=4,
+    help="The number of ways in the cache to partition to the default "
+    "PartitionID when using Way Cache Partitioning Policy",
+)
+
+args = parser.parse_args()
+
+
+m5.ticks.setGlobalFrequency(STATIC_OPTIONS.system_frequency)
+system = configSystem()
+
+# create a cache to sit between to enforce Partitioning policies
+PartitionPolicy = generatePartPolicy(args)
```

Comment:
ops I missed this. PartitionPolicy should be lowercase with underscore, otherwise it might be confused for a type:

partition_policy

Reply:
Agreed, I've renamed it as suggested